RHDEVDOCS-3824 - Added setting up argo cd instance (#46559)
openshift/openshift-docs: cicd/gitops/setting-up-argocd-instance.adoc

:_content-type: ASSEMBLY
[id="setting-up-argocd-instance"]
= Setting up a new Argo CD instance
include::_attributes/common-attributes.adoc[]
:context: setting-up-argocd-instance

toc::[]

> By default, the {gitops-title} installs an instance of Argo CD in the `openshift-gitops` namespace with additional permissions for managing certain cluster-scoped resources. To manage cluster configurations or deploy applications, you can install and deploy a new Argo CD instance. By default, any new instance has permissions to manage resources only in the namespace where it is deployed.
> 
> 
> include::modules/gitops-argo-cd-installation.adoc[leveloffset=+1]

Reviewer: ```suggestion
By default, the {gitops-title} installs an instance of Argo CD in the `openshift-gitops` namespace with additional permissions for managing certain cluster-scoped resources. To manage cluster configurations or deploy applications, you can install and deploy a new Argo CD instance. By default, any new instance has permissions to manage resources only in the namespace where it is deployed.

include::modules/gitops-argo-cd-installation.adoc[leveloffset=+1]
```

Minor nit, not a blocker but I'd remove the unnecessary extra line.

:_content-type: ASSEMBLY
[id="setting-up-argocd-instance"]
= Setting up a new Argo CD instance
include::_attributes/common-attributes.adoc[]
:context: setting-up-argocd-instance

toc::[]

By default, the {gitops-title} installs an instance of Argo CD in the `openshift-gitops` namespace with additional permissions for managing certain cluster-scoped resources.

To manage cluster configurations or deploy applications, you can install and deploy a new Argo CD instance. By default, any new instance has permissions to manage resources only in the namespace where it is deployed.

> 

Reviewer: Make these two sentences one paragraph, then it is more closely aligned to the upcoming Jupiter requirements for abstracts:
https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=JUP&title=Jupiter+Source+files+readiness+Checklist#JupiterSourcefilesreadinessChecklist-ContentChecklist

:_content-type: ASSEMBLY
[id="setting-up-argocd-instance"]
= Setting up a new Argo CD instance
include::_attributes/common-attributes.adoc[]
:context: setting-up-argocd-instance

toc::[]

> The {gitops-title} installs an instance of Argo CD in `openshift-gitopsnamespace` with additional permissions, which allows to manage certain cluster scoped resources.

Reviewer 2: ```suggestion
By default, the {gitops-title} installs an instance of Argo CD in the `openshift-gitops`namespace with additional permissions, which allows to manage certain cluster scoped resources.
```

> By default, the {gitops-title} installs an instance of Argo CD in the `openshift-gitops` namespace with additional permissions, which allows to manage certain cluster scoped resources.

Reviewer 3: @DebarghoGhosh please review this suggestion.

> You can install and deploy a new Argo CD instance for managing cluster configurations or deploying applications. By default, the new instance will only have permissions to manage resources in the namespace where it is deployed.

Reviewer 2: ```suggestion
You can install and deploy a new Argo CD instance for managing cluster configurations or deploying applications. By default, any new instance will only have permissions to manage resources in the namespace where it is deployed.
```

> You can install and deploy a new Argo CD instance for managing cluster configurations or deploying applications. By default, any new instance will only have permissions to manage resources in the namespace where it is deployed.

Reviewer 3: ```suggestion
To manage cluster configurations or deploy applications, you can install and deploy a new Argo CD instance. By default, any new instance only has permissions to manage resources in the namespace where it is deployed.
```
- Where possible, avoid the future-tense. Use the present tense instead. 
- What is the intention of this paragraph? Are you describing the user's motivation for (re)installing Argo CD instead of using the default installation? If so, you can emphasize that motivation by putting it first.

include::modules/gitops-argo-cd-installation.adoc[leveloffset=+1]

include::modules/gitops-enable-replicas-for-argo-cd-server.adoc[leveloffset=+1]

include::modules/gitops-deploy-resources-different-namespaces.adoc[leveloffset=+1]

> ////

Reviewer 3: It's unclear whether this Additional resources section is for the assembly or the module referenced by the include statement on line 18.

If it is for the assembly: 
- Use `== Additional resources` instead of `.Additional resources`.
- Specify an ID [as described here](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies).

Reviewer 3: Do you intend to keep the `////` comment block?
